parser/modify_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestPatchList(t *testing.T) {
+	in := []byte("abcdefghijklmnopqrstuvwxyz")
+
+	patchlist := PatchList{}
+	err := patchlist.Add(0, 3, "ABC")
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = patchlist.Add(12, 15, "MNO")
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = patchlist.Add(24, 26, "Z")
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = patchlist.Add(15, 15, "_")
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = patchlist.Add(12, 13, "x")
+	if _, ok := err.(PatchOverlapError); !ok {
+		t.Error("missing PatchOverlapError")
+	}
+
+	err = patchlist.Add(13, 14, "x")
+	if _, ok := err.(PatchOverlapError); !ok {
+		t.Error("missing PatchOverlapError")
+	}
+
+	err = patchlist.Add(14, 15, "x")
+	if _, ok := err.(PatchOverlapError); !ok {
+		t.Error("missing PatchOverlapError")
+	}
+
+	err = patchlist.Add(11, 13, "x")
+	if _, ok := err.(PatchOverlapError); !ok {
+		t.Error("missing PatchOverlapError")
+	}
+
+	err = patchlist.Add(12, 15, "x")
+	if _, ok := err.(PatchOverlapError); !ok {
+		t.Error("missing PatchOverlapError")
+	}
+
+	err = patchlist.Add(11, 15, "x")
+	if _, ok := err.(PatchOverlapError); !ok {
+		t.Error("missing PatchOverlapError")
+	}
```

Comment:
Can you test `patchlist.Add(15, 15, "x")`? That should return a PatchOverlapError, but if I'm reading the conditional correctly, will apply without error.

Reply:
Sorry, my complaint was that we weren't catching overlapping single character patches. Since we already did one patch on line 41, doing it again should fail with an error. But since we're doing a `<` in every section of the conditional, and both starts and ends are equal, we'll never error.

Reply:
Done